### PR TITLE
CI with GitHub actions

### PR DESCRIPTION
**Pull request by dmgav**
Convert CI from Trevis to GitHub actions. Minor fixes in unit tests to make them run on Windows.

### CODE REVIEW

**Review by mrakitin — COMMENTED** [review]
Looks good overall. I have a couple of questions (see below).

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  push:
+  pull_request:
+
+jobs:
+
+  build:
+    env:
+      DISPLAY: ":99.0"  # arbitrary
+      # GSETTINGS_SCHEMA_DIR_CONDA_BACKUP is somehow related to the following issue https://github.com/conda/conda/issues/9966
+      GSETTINGS_SCHEMA_DIR_CONDA_BACKUP: "/usr/share/miniconda/envs/testenv/share/glib-2.0/schemas"
```
[the remark this answers is not in the document]
Comment (dmgav) [reply .github/workflows/testing.yml:13]: I think the error message was `GSETTINGS_SCHEMA_DIR_CONDA_BACKUP` is not bound`. Pretty much how it is described in the referenced issue.

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      matrix:
+        host-os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        python-version: [3.7, 3.8, 3.9]
+        numpy-version: ["1.20"]
+        pyqt-version: ["5.12"]
+      fail-fast: false
+
+    runs-on: ${{ matrix.host-os }}
+
+    # Note: Services are based on docker contains and would run only on Linux
```
[the remark this answers is not in the document]
Comment (dmgav) [reply .github/workflows/testing.yml:25]: containers

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      run: |
+        sudo /sbin/start-stop-daemon --start --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -screen 0 1920x1200x24 -ac +extension GLX +render -noreset
+        # sudo /usr/bin/Xvfb $DISPLAY -screen 0 1280x1024x24 &
+        sleep 3
+
+    - name:  Start 'herbstluftwm'
+      if: runner.os == 'Linux'
+      run: |
+        herbstluftwm &
+        sleep 1
```
[the remark this answers is not in the document]
Comment (dmgav) [reply .github/workflows/testing.yml:74]: You can still run tests with limited functionality with `xvfb`, but you need `herbstluftwm` to model things like button clicks (click button and receive an event). The configuration pretty much follows this: https://github.com/slaclab/pydm/blob/8752696d5b2078784725e7f70cc0e6fcb8a882a4/azure-test-template.yml#L29

[the remark this answers is not in the document]
Comment (mrakitin) [reply .github/workflows/testing.yml:74]: Nice. Do you run the click tests for this project? Or is it a preparation for the future tests?

File: .github/workflows/testing.yml (hunk 2 above)
Comment (mrakitin) [inline .github/workflows/testing.yml:25]:
```suggestion
    # Note: Services are based on docker containers and would run only on Linux
```

File: .github/workflows/testing.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (dmgav) [reply .github/workflows/testing.yml:74]: I don't remember, but probably yes. Also it's a good idea to have all the projects configured identically.

File: .github/workflows/testing.yml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      if: runner.os == 'Linux'
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y xvfb herbstluftwm
+
+    - name: Start 'xvfb'
+      if: runner.os == 'Linux'
+      run: |
+        sudo /sbin/start-stop-daemon --start --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -screen 0 1920x1200x24 -ac +extension GLX +render -noreset
+        # sudo /usr/bin/Xvfb $DISPLAY -screen 0 1280x1024x24 &
```
Comment (dmgav) [inline .github/workflows/testing.yml:67]:
```suggestion
```

**Pull request merged**